src/multivariate/product.jl

```diff
@@ -39,7 +39,7 @@ end
 _rand!(rng::AbstractRNG, d::Product, x::AbstractVector{<:Real}) =
     map!(Base.Fix1(rand, rng), x, d.v)
 _logpdf(d::Product, x::AbstractVector{<:Real}) =
-    sum(n->logpdf(d.v[n], x[n]), 1:length(d))
+    sum(n->logpdf(d.v[n], x[n]), 1:length(d); init=0.0)
```

Comment:
Probably we rather want something like
```suggestion
function _logpdf(d::Product, x::AbstractVector{<:Real})
    if isempty(dists)
        return sum(map(logpdf, d.v, x))
    end
    dists = d.v
    return sum(n -> logpdf(dists[n], x[n]), 1:length(d))
end
```